third_party/move/move-compiler-v2/src/acquires_checker.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Performs strict acquire analysis as outlined in the move book, that is:
+//! A Move function `m::f` must be annotated with acquires `T`` if and only if,
+//! - The body of `m::f` contains a `move_from<T>`, `borrow_global_mut<T>`, or `borrow_global<T>` instruction, or
+//! - The body of `m::f` invokes a function `m::g` declared in the same module that is annotated with acquires
+//! Warn if access specifiers other than plain `acquire R` is used.
+//! This check is enabled by flag `Experiment::ACQUIRES_CHECK`, and is disabled by default.
+
+use move_binary_format::file_format;
+use move_model::{
+    ast::{AddressSpecifier, ExpData, Operation, ResourceSpecifier},
+    model::{FunId, FunctionEnv, GlobalEnv, Loc, ModuleEnv, StructId},
+    ty::Type,
+};
+use std::collections::{BTreeMap, BTreeSet, VecDeque};
+
+/// Performs acquire checking
+pub fn acquires_checker(env: &GlobalEnv) {
+    for module in env.get_modules() {
+        if module.is_target() {
+            let analyzer = AcquireChecker::new(module.clone());
+            let acquires = analyzer.analyze();
+            for (fun_id, acquires) in acquires.into_iter() {
+                let fun_env = module.get_function(fun_id);
+                let mut declared_acquires = get_acquired_resources(&fun_env);
+                for (sid, acquired) in acquires.0 {
+                    if declared_acquires.remove(&sid).is_none() {
+                        let s_name = module.get_struct(sid).get_name();
+                        let note = match acquired {
+                            AcquiredAt::Directly(loc) => (loc, "acquired here".to_owned()),
+                            AcquiredAt::Indirectly(loc, _fun_id) => {
+                                (loc, "acquired by call".to_owned())
```

Comment:
As you have `_fun_id` here, perhaps print the simple function name (`env.get_function(module.get_id().qualified(fun_id).get_name().display(...)`)

Reply:
done